packages/sources/the-network-firm/src/transport/reserve.ts

```diff
@@ -50,34 +54,52 @@ export const httpTransport = new HttpTransport<HttpTransportTypes>({
   },
   parseResponse: (params, response) => {
     return params.map((param) => {
+      const laxRipcord = param.laxRipcord
+
       const ripcord =
         response.data.ripcord || response.data.ripcord.toString().toLowerCase() === 'true'
+      let ripcordDetails = ''
+      const ripcordAsInt = ripcord ? 1 : 0
+      const totalReserve = Number(ripcord ? 0 : response.data.totalReserve)
+      const totalToken = Number(ripcord ? 0 : response.data.totalToken)
+
+      // Populate ripcord details if ripcord is true
       if (ripcord) {
-        const message = `Ripcord indicator true. Details: ${response.data.ripcordDetails.join(
-          ', ',
-        )}`
-        return {
-          params: param,
-          response: {
-            errorMessage: message,
-            ripcord: response.data.ripcord,
-            ripcordDetails: response.data.ripcordDetails.join(', '),
-            statusCode: 502,
-            timestamps: {
-              providerIndicatedTimeUnixMs: new Date(response.data.timestamp).getTime(),
+        ripcordDetails = response.data.ripcordDetails.join(', ')
+
+        // If laxRipcord is false and ripcord is true return 502
+        if (!laxRipcord) {
+          const message = `Ripcord indicator true. Details: ${response.data.ripcordDetails.join(
+            ', ',
+          )}`
+          return {
+            params: param,
+            response: {
+              errorMessage: message,
+              ripcord: response.data.ripcord,
+              ripcordAsInt,
+              ripcordDetails,
+              statusCode: 502,
+              timestamps: {
+                providerIndicatedTimeUnixMs: new Date(response.data.timestamp).getTime(),
+              },
             },
-          },
+          }
         }
+
+        logger.debug(`Ripcord indicator true. Details: ${ripcordDetails}`)
       }
 
-      const result = response.data
       return {
         params: param,
         response: {
-          result: result.totalReserve,
+          result: totalReserve,
           data: {
-            result: result.totalReserve,
+            result: totalReserve,
             ripcord: response.data.ripcord,
+            ripcordAsInt,
+            totalReserve,
+            totalToken,
```

Comment:
i added both of these as named responses to make it easier to understand whats going on but i could remove it - i dont think we'll need them - if you think i should remove this let me know